### ISSUE TITLE
Support for inline math using LaTeX

### ISSUE THREAD

**Issue by Yeeler** [vuejs/vuepress#289]
I have quite lot of markdown files which contain many latex equations.
How can I get them show up properly in vuepress?

e.g.
$J \left( \theta_0, \theta_1 \right) = \frac{1}{2m}\sum\limits_{i=1}^m \left( h_{\theta}(x^{(i)})-y^{(i)} \right)^{2}$

**Comment by mtobeiyf** [comment]
You can customize by yourself. 
There are instructions here 👉  https://github.com/vuejs/vuepress/issues/113

**Comment by Yeeler** [comment]
Latex has different syntax from Katex. Markdown-it-katex didn't work on latex. I tried many other markdown plugins, including markdown-it-latex, markdown-it-mathjax, amd markdown-it-texmath etc. No luck. 

When I was previewing markdown files in VS Code, I can see the above equation with a plug-in named 'markdown-math' installed. Why l can't do the same in Vuepress?

My current solution is using https://www.codecogs.com/eqnedit.php to insert an equation image. Is there any better choice? How can I show latex equation in node JS? What package should I use?

**Comment by mtobeiyf** [comment]
[Katex](https://github.com/Khan/KaTeX) is a library for TeX math rendering. 

Following the guide in #113 , you'll get your inline math equation parsed and rendered correctly.

![image](https://user-images.githubusercontent.com/5097752/39404694-f8cd4300-4bca-11e8-84a3-f02187b05afe.png)

For all the functions supported in Katex, you may check its official doc [KaTeX/function-support](https://khan.github.io/KaTeX/function-support.html)

It's enough for you to handle 99% of the cases like the cost function shown above.

**Comment by Yeeler** [comment]
I don't know how to add Katex CSS into my theme, but I did all the other steps. And I got
![image](https://user-images.githubusercontent.com/5031611/39405171-e351c606-4bd2-11e8-9323-4b1bc34c9d2d.png)

this is my markdown file:
`# Latex demo

$J \left( \theta_0, \theta_1 \right) = \frac{1}{2m}\sum\limits_{i=1}^m \left( h_{\theta}(x^{(i)})-y^{(i)} \right)^{2}$`

Can you please provide your vuepress project? Thanks for your help.

**Comment by Yeeler** [comment]
you are right. I missed the css. I didn't know that we can apply css in markdown file in this way.

I copied default theme by run 'vuepress eject docs'
then I modified 'D:\_GIT\vue01\docs\.vuepress\theme\Layout.vue'
append '<style src="markdown-it-katex/node_modules/katex/dist/katex.min.css"></style>'

finally, all worked.

**Comment by ulivz** [comment]
See: https://github.com/vuejs/vuepress/issues/113#issuecomment-382669955

Closing as this can be done in userland.

**Comment by Yeeler** [comment]
yeah I like it.

**Comment by faroit** [comment]
I actually couldn't find the info that its possible to add custom headers without ejecting the theme. This is great! Maybe this could go into the documentation!

**Comment by RyanGreenup** [comment]
If anybody else is facing a similar problem and really wants to use MathJax rather than KaTeX (in order to support `\label{}` in an `align` environment rather than merely `\tag{]` in an `\[ aligned \]` environment for instance) a similar thing can be done by injecting the [*cdnjs* MathJax script](https://docs.mathjax.org/en/v2.7-latest/start.html):

```HTML
<script type="text/javascript" async
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML">
</script>
```

Which can be acheived in *VuePress*, as described in the [manual](https://vuepress.vuejs.org/config/#head) by adding the following to the `.vuepress/config.js`:

```javascript
head: [
  ['script',
    { type: '"text/javascript" async',
    src: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML'
    }
  ]
],
```

This wouldn't be worth adding to this issue, but the new community [`vuepress-plugin-mathjax`](https://vuepress.github.io/en/plugins/mathjax/#installation) plugin does not render LaTeX in an aligned environment and  the deprecated [`latex`](https://www.npmjs.com/package/vuepress-plugin-latex) plugin doesn't quite render the math in a way that looks correct:
```latex
$$
\begin{aligned}
         0 &= ax^2 +bx +c
\implies x &= \frac{- b \pm \sqrt{b^2- 4ac} }{2a} 
\end{aligned}
$$
```
This method was suggestested but not detailed [in #113 by songololo](https://github.com/vuejs/vuepress/issues/113#issuecomment-466984783).

**Comment by BruceYuj** [comment]
@RyanGreenup  I use your method to fix some syntax like `$$\begin{cases} \end{cases}$$`, but it didn't work. I use the community plugin - `vuepress-plugin-mathjax` and manually inject Mathjax script. Can you help me?
